src/driver.rs

```diff
@@ -59,20 +63,51 @@ fn test_arg_value() {
     assert_eq!(arg_value(args, "--foo", |_| true), None);
 }
 
+fn track_clippy_args(sess: &Session, args_env_var: &Option<String>) {
+    sess.parse_sess.env_depinfo.borrow_mut().insert((
+        Symbol::intern("CLIPPY_ARGS"),
+        args_env_var.as_deref().map(Symbol::intern),
+    ));
+}
+
 struct DefaultCallbacks;
 impl rustc_driver::Callbacks for DefaultCallbacks {}
 
-struct ClippyCallbacks;
+struct ClippyArgsCallbacks {
+    clippy_args_var: Option<String>,
+}
+
+impl rustc_driver::Callbacks for ClippyArgsCallbacks {
+    fn config(&mut self, config: &mut interface::Config) {
+        let previous = config.register_lints.take();
+        let clippy_args_var = self.clippy_args_var.take();
+        config.register_lints = Some(Box::new(move |sess, lint_store| {
```

Comment:
I'm not sure if this is the correct place to modify `env_depinfo`... It works, but the callback is called `register_lints`.

Reply:
This is indeed abusing the `register_lints` callback. I'm not familiar with the clippy driver stuff. I also don't know who to ping for advice here. @oli-obk do you have any pointers here? Maybe we should add a new callback to config to address this issue?

Reply:
imo this is fine, but we should look into figuring out the story around adding custom things to the depinfo in general. This seems like a larger topic (that may want to include proc macros and similar).

I think we should merge this as it is and then (once clippy is synced into the rustc repo) consider doing a change there that adds a `parse_sess_created` callback that allows you to modify the session right after it was created.